geode-core/src/main/java/org/apache/geode/management/internal/cli/domain/RegionConfigFactory.java

```diff
@@ -32,310 +32,255 @@
+      regionAttributesType.setConcurrencyLevel(args.getConcurrencyLevel().toString());
     } else if (regionAttributes != null) {
-      addAttribute(regionConfig, a -> a.setConcurrencyLevel(Integer.toString(
-          regionAttributes.getConcurrencyLevel())));
+      regionAttributesType
+          .setConcurrencyLevel(Integer.toString(regionAttributes.getConcurrencyLevel()));
     }
 
     if (regionAttributes != null && regionAttributes.getDataPolicy() != null) {
-      addAttribute(regionConfig,
-          a -> a.setDataPolicy(regionAttributes.getDataPolicy().toConfigType()));
+      regionAttributesType.setDataPolicy(regionAttributes.getDataPolicy().toConfigType());
     }
 
     if (regionAttributes != null && regionAttributes.getScope() != null
         && !regionAttributes.getDataPolicy().withPartitioning()) {
-      addAttribute(regionConfig,
-          a -> a.setScope(
-              RegionAttributesScope.fromValue(regionAttributes.getScope().toConfigTypeString())));
+      regionAttributesType.setScope(
+          RegionAttributesScope.fromValue(regionAttributes.getScope().toConfigTypeString()));
     }
 
     return regionConfig;
   }
 
-  private String int2string(Integer i) {
-    return Optional.ofNullable(i).map(j -> j.toString()).orElse(null);
+  public static RegionAttributesType.ExpirationAttributesType getExpirationAttributes(
+      ExpirationAttributes entryIdleTimeout, CustomExpiry<?, ?> customEntryIdleTimeout) {
+
+    if ((entryIdleTimeout == null || entryIdleTimeout.isDefault())
+        && customEntryIdleTimeout == null) {
+      return null;
+    }
+
+    if (entryIdleTimeout == null || entryIdleTimeout.isDefault()) {
+      return getExpirationAttributes(null, null,
+          new ClassName<>(customEntryIdleTimeout.getClass().getName()));
+    } else if (customEntryIdleTimeout == null) {
+      return getExpirationAttributes(entryIdleTimeout.getTimeout(), entryIdleTimeout.getAction(),
+          null);
+    } else {
+      return getExpirationAttributes(entryIdleTimeout.getTimeout(), entryIdleTimeout.getAction(),
+          new ClassName<>(customEntryIdleTimeout.getClass().getName()));
+    }
   }
 
-  private String long2string(Long i) {
-    return Optional.ofNullable(i).map(j -> j.toString()).orElse(null);
+  public static RegionAttributesType.ExpirationAttributesType getExpirationAttributes(
+      RegionFunctionArgs.ExpirationAttrs expirationAttrs, ClassName<CustomExpiry> customExpiry) {
+    if (expirationAttrs == null) {
+      return getExpirationAttributes(null, null, customExpiry);
+    } else {
+      return getExpirationAttributes(expirationAttrs.getTime(), expirationAttrs.getAction(),
+          customExpiry);
+    }
   }
 
-  private String getLeafRegion(String fullPath) {
-    String regionPath = fullPath;
-    String[] regions = regionPath.split("/");
+  public static RegionAttributesType.ExpirationAttributesType getExpirationAttributes(
```

Comment:
I thought I added a helper function in ExpirationAttributes.java to do this? Should it not be under that domain?

Reply:
this take CustomExpiry into consideration as well. ExpirationAttributes only has timeout and action.

Reply:
Okay.